### DISCUSSION TITLE
Problem in installing the new version

### DISCUSSION THREAD

**Discussion by optibikash** [pybamm-team/PyBaMM#2003] [General]
I am unable to install the new version of PyBamm. 
I used the basic command and got the following error:
`PS C:\Users\bsah\ownCloud-TREE\Bikash\Other works\Python_works> pip install pybamm
Collecting pybamm
  Using cached pybamm-0.2.4.post3-py3-none-any.whl (520 kB)
Collecting scipy>=1.3
  Using cached scipy-1.8.0-cp310-cp310-win_amd64.whl (37.0 MB)
Collecting autograd>=1.2
  Using cached autograd-1.3.tar.gz (38 kB)
  Preparing metadata (setup.py) ... done
Collecting anytree>=2.4.3
  Using cached anytree-2.8.0-py2.py3-none-any.whl (41 kB)
Collecting pybamm
  Using cached pybamm-0.2.4.post2-py3-none-any.whl (520 kB)
  Using cached pybamm-0.2.4.post1-py3-none-any.whl (520 kB)
  Using cached pybamm-0.2.4-py3-none-any.whl (520 kB)
  Using cached pybamm-0.2.3.post1.tar.gz (281 kB)
  Preparing metadata (setup.py) ... done
Collecting numpy>=1.16
  Using cached numpy-1.22.3-cp310-cp310-win_amd64.whl (14.7 MB)
Collecting pandas>=0.24
  Using cached pandas-1.4.1-cp310-cp310-win_amd64.whl (10.6 MB)
Collecting scikit-fem>=0.2.0
  Using cached scikit_fem-6.0.0-py3-none-any.whl (144 kB)
Collecting pybamm
  Using cached pybamm-0.2.3-py3-none-any.whl (485 kB)    
  Using cached pybamm-0.2.2.post3-py3-none-any.whl (450 kB)
  Using cached pybamm-0.2.2.post2-py3-none-any.whl (448 kB)
  Using cached pybamm-0.2.2.post1-py3-none-any.whl (446 kB)
  Using cached pybamm-0.2.2-py3-none-any.whl (444 kB)
  Using cached pybamm-0.2.1-py3-none-any.whl (402 kB)
  Using cached pybamm-0.2.0-py3-none-any.whl (371 kB)
  Using cached pybamm-0.1.0-py3-none-any.whl (310 kB)
ERROR: Cannot install pybamm==0.1.0, pybamm==0.2.0, pybamm==0.2.1, pybamm==0.2.2, pybamm==0.2.2.post1, pybamm==0.2.2.post2, pybamm==0.2.2.post3, pybamm==0.2.3, pybamm==0.2.3.post1, pybamm==0.2.4, pybamm==0.2.4.post1, pybamm==0.2.4.post2 and pybamm==0.2.4.post3 because these package versions have conflicting dependencies.

The conflict is caused by:
    pybamm 0.2.4.post3 depends on casadi>=3.5.0
    pybamm 0.2.4.post2 depends on casadi>=3.5.0
    pybamm 0.2.4.post1 depends on casadi>=3.5.0
    pybamm 0.2.4 depends on casadi>=3.5.0
    pybamm 0.2.3.post1 depends on casadi>=3.5.0
    pybamm 0.2.3 depends on casadi>=3.5.0
    pybamm 0.2.2.post3 depends on casadi>=3.5.0
    pybamm 0.2.2.post2 depends on casadi>=3.5.0
    pybamm 0.2.2.post1 depends on casadi>=3.5.0
    pybamm 0.2.2 depends on casadi>=3.5.0
    pybamm 0.2.1 depends on casadi>=3.5.0
    pybamm 0.2.0 depends on casadi>=3.5.0
    pybamm 0.1.0 depends on casadi>=3.5.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
PS C:\Users\bsah\ownCloud-TREE\Bikash\Other works\Python_works> `

Later I tried by loosening the range but it too didn't worked. 
Need suggestions for installation!

**Comment by actualpriyanshu** [comment]
We strongly recommend to install PyBaMM within a python virtual environment, in order not to alter any distribution python files. For instructions on how to create a virtual environment for PyBaMM, see [the documentation](https://pybamm.readthedocs.io/en/latest/install/GNU-linux.html#user-install).

Furthermore, you are trying to install pybamm v0.1 and v0.2. To install the latest version (v22.3) use
`pip install pybamm==22.3`

**Reply by optibikash** [reply]
Thanks. The problem was not with the virtual environment but rather the python version. I tried using the latest one 3.10.4. The issue was solved as soon as I got back to 3.9. I should have checked the documentation.

**Reply by brosaplanella** [reply]
At the moment we do not support Python 3.10, hence the error when installing.
